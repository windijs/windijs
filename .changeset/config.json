{
  "$schema": "https://unpkg.com/@changesets/config@2.1.1/schema.json",
  "changelog": "@changesets/cli/changelog",
  "commit": false,
  "fixed": [["@windijs/core", "windijs"]],
  "linked": [],
  "access": "public",
  "baseBranch": "main",
  "updateInternalDependencies": "patch",
  "ignore": []
}
