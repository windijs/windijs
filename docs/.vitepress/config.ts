import { defineConfig } from "vitepress";
import { mdRenderFilename } from "./markdown";

export default defineConfig({
  lang: "en-US",
  title: "Windi JS",
  cleanUrls: "without-subfolders",
  description: "The Utility-first CSS-in-JS Framework Powered by ES6 Proxy API and TypeScript.",
  markdown: {
    config(md) {
      md.use(mdRenderFilename);
    },
  },
  head: [
    ['link', { rel: "shortcut icon", type: "image/x-icon", href: "/windijs/favicon.ico" }]
  ],
  base: "/windijs/",
  themeConfig: {
    logo: "/logo.png",
    nav: [
      { text: "Guide", link: "/guide/what-is-windijs" },
      { text: "API", link: "/api" },
      { text: "Repl", link: "/repl" },
      { text: "Examples", link: "https://github.com/windijs/windijs/tree/main/examples" },
    ],
    sidebar: [
      {
        text: "Guide",
        collapsible: false,
        items: [
          {
            text: "What is Windi JS?",
            link: "/guide/what-is-windijs",
          },
          {
            text: "Getting Started",
            link: "/guide/getting-started",
          },
          {
            text: "Configuration",
            link: "/guide/configuration",
          },
          {
            text: "Optimization",
            link: "/guide/optimization",
          },
        ],
      },
      {
        text: "Customization",
        collapsible: false,
        items: [
          {
            text: "Theme",
            link: "/customization/theme",
          },
          {
            text: "Utility",
            link: "/customization/utility",
          },
          {
            text: "Variant",
            link: "/customization/variant",
          },
          {
            text: "Handler",
            link: "/customization/handler",
          },
          {
            text: "Loader",
            link: "/customization/loader",
          },
          {
            text: "Namer",
            link: "/customization/namer",
          },
          {
            text: "Plugins",
            link: "/customization/plugins",
          },
        ],
      },
      {
        text: "Integrations",
        collapsible: false,
        items: [
          {
            text: "API",
            link: "/integrations/api",
          },
          // {
          //   text: "CLI",
          //   link: "/integrations/cli",
          // },
          {
            text: "Vite",
            link: "/integrations/vite",
          },
          // {
          //   text: "Rollup",
          //   link: "/integrations/rollup",
          // },
          // {
          //   text: "Webpack",
          //   link: "/integrations/webpack",
          // },
        ],
      },
      {
        text: "Posts",
        collapsible: false,
        items: [
          {
            text: "How Windi JS Works",
            link: "/posts/how-windijs-works",
          },
          {
            text: "Ways of Using Windi JS",
            link: "/posts/ways-of-using-windijs",
          },
        ],
      },
    ],
    editLink: {
      pattern: "https://github.com/windijs/windijs/edit/main/docs/:path",
      text: "Edit this page on GitHub",
    },
    socialLinks: [
      { icon: "discord", link: "https://discord.gg/nJux6Gb52m" },
      { icon: "github", link: "https://github.com/windijs/windijs" },
    ],
    footer: {
      message: "Released under the MIT License.",
      copyright: "Copyright Â© 2022-present Raven Satir",
    },
  },
});
