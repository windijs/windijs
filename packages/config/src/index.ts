export * from "./bg";
export * from "./svg";
export * from "./flex";
export * from "./grid";
export * from "./border";
export * from "./effects";
export * from "./behaviors";
export * from "./filter";
export * from "./opacity";
export * from "./sizing";
export * from "./spacing";
export * from "./animation";
export * from "./transform";
export * from "./positioning";
export * from "./interactivity";
export * from "./transitions";
export * from "./typography";
export * from "./keyframes";
export * from "./interface";
