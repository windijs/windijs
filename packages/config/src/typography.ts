import { range, spacings } from "@windijs/shared";

export const fontFamilyConfig = {
  sans: [
    "ui-sans-serif",
    "system-ui",
    "-apple-system",
    "BlinkMacSystemFont",
    "\"Segoe UI\"",
    "Roboto",
    "\"Helvetica Neue\"",
    "Arial",
    "\"Noto Sans\"",
    "sans-serif",
    "\"Apple Color Emoji\"",
    "\"Segoe UI Emoji\"",
    "\"Segoe UI Symbol\"",
    "\"Noto Color Emoji\"",
  ],
  serif: [
    "ui-serif",
    "Georgia",
    "Cambria",
    "\"Times New Roman\"",
    "Times",
    "serif",
  ],
  mono: [
    "ui-monospace",
    "SFMono-Regular",
    "Menlo",
    "Monaco",
    "Consolas",
    "\"Liberation Mono\"",
    "\"Courier New\"",
    "monospace",
  ],
};

export const fontStyleConfig = {
  italic: "italic",
  normal: "normal",
};

export const fontSizeConfig = {
  xs: ["0.75rem", { lineHeight: "1rem" }],
  sm: ["0.875rem", { lineHeight: "1.25rem" }],
  base: ["1rem", { lineHeight: "1.5rem" }],
  lg: ["1.125rem", { lineHeight: "1.75rem" }],
  xl: ["1.25rem", { lineHeight: "1.75rem" }],
  "2xl": ["1.5rem", { lineHeight: "2rem" }],
  "3xl": ["1.875rem", { lineHeight: "2.25rem" }],
  "4xl": ["2.25rem", { lineHeight: "2.5rem" }],
  "5xl": ["3rem", { lineHeight: "1" }],
  "6xl": ["3.75rem", { lineHeight: "1" }],
  "7xl": ["4.5rem", { lineHeight: "1" }],
  "8xl": ["6rem", { lineHeight: "1" }],
  "9xl": ["8rem", { lineHeight: "1" }],
};

export const fontWeightConfig = {
  thin: "100",
  extralight: "200",
  light: "300",
  normal: "400",
  medium: "500",
  semibold: "600",
  bold: "700",
  extrabold: "800",
  black: "900",
};

export const hyphensConfig = {
  auto: "auto",
  manual: "manual",
  none: "none",
};

export const letterSpacingConfig = {
  tighter: "-0.05em",
  tight: "-0.025em",
  normal: "0em",
  wide: "0.025em",
  wider: "0.05em",
  widest: "0.1em",
};

export const lineHeightConfig = {
  none: "1",
  tight: "1.25",
  snug: "1.375",
  normal: "1.5",
  relaxed: "1.625",
  loose: "2",
  ...spacings(range(3, 11)),
};

export const tabSizeConfig = {
  DEFAULT: "4",
  0: "0",
  2: "2",
  4: "4",
  8: "8",
};

export const textAlignConfig = {
  left: "left",
  center: "center",
  right: "right",
  justify: "justify",
};

export const verticalAlignConfig = {
  baseline: "baseline",
  top: "top",
  middle: "middle",
  bottom: "bottom",
  textTop: "text-top",
  textBottom: "text-bottom",
};

export const fontVariantNumericConfig = {
  nums: {
    normal: "normal",
    ordinal: "ordinal",
    slashedZero: "slashed-zero",
    lining: "lining-nums",
    oldstyle: "oldstyle-nums",
    proportional: "proportional-nums",
    tabular: "tabular-nums",
  },
  fractions: {
    diagonal: "diagonal-fractions",
    stacked: "stacked-fractions",
  },
};

export const textDecorationTypeConfig = {
  underline: "underline",
  overline: "overline",
  linethrough: "line-through",
  none: "none",
};

export const textDecorationStyleConfig = {
  solid: "solid",
  double: "double",
  dotted: "dotted",
  dashed: "dashed",
  wavy: "wavy",
};

export const textDecorationThicknessConfig = {
  auto: "auto",
  0: "0px",
  1: "1px",
  2: "2px",
  3: "3px",
  4: "4px",
  5: "5px",
  6: "6px",
  7: "7px",
  8: "8px",
};

export const textDecorationOffsetConfig = textDecorationThicknessConfig;

export const textIndentConfig = {
  DEFAULT: "1.5rem",
  xs: "0.5rem",
  sm: "1rem",
  md: "1.5rem",
  lg: "2rem",
  xl: "2.5rem",
  xxl: "3rem",
  xxxxl: "4rem",
};

export const textStrokeWidthConfig = {
  DEFAULT: "medium",
  none: "0",
  sm: "thin",
  md: "medium",
  lg: "thick",
};

export const textShadowConfig = {
  DEFAULT: "0px 0px 1px rgb(0 0 0 / 20%), 0px 0px 1px rgb(1 0 5 / 10%)",
  sm: "1px 1px 3px rgb(36 37 47 / 25%)",
  md: "0px 1px 2px rgb(30 29 39 / 19%), 1px 2px 4px rgb(54 64 147 / 18%)",
  lg: "3px 3px 6px rgb(0 0 0 / 26%), 0 0 5px rgb(15 3 86 / 22%)",
  xl: "1px 1px 3px rgb(0 0 0 / 29%), 2px 4px 7px rgb(73 64 125 / 35%)",
  none: "none",
};

export const textTransformConfig = {
  normal: "none",
  uppercase: "uppercase",
  lowercase: "lowercase",
  capitalize: "capitalize",
};

export const whiteSpaceConfig = {
  normal: "normal",
  nowrap: "nowrap",
  pre: {
    DEFAULT: "pre",
    line: "pre-line",
    wrap: "pre-wrap",
  },
};

export const writingModeConfig = {
  normal: "horizontal-tb",
  vertical: {
    right: "vertical-rl",
    left: "vertical-lr",
  },
};

export const writingOrientationConfig = {
  mixed: "mixed",
  upright: "upright",
  sideways: "sideways",
};
