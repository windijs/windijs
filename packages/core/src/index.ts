export * from "./base";
export * from "./api";
export * from "./handler";
export * from "./prefixer";
export * from "./builder";
