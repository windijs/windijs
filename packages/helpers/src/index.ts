// TODO: maybe add more helpers reference sass?
// https://sass-lang.com/documentation/modules

export * from "./types";
export * from "./build";
export * from "./config";
export * from "./color";
export * from "./common";
export * from "./css";
export * from "./data";
export * from "./dollar";
export * from "./escape";
export * from "./funcs";
export * from "./math";
export * from "./meta";
export * from "./namer";
export * from "./proxy";
export * from "./unit";
