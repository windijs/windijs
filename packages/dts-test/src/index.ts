import { defineConfig } from "@windijs/config";

defineConfig({
  utilities: {

  },
});
