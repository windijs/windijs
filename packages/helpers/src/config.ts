import { Handler, StyleLoader, StyleObject } from "./types";

import { Theme } from "@windijs/config";

export type DarkModeConfig = "class" | "media" | false;

export type UtilitiesConfig = { [key: string]: Handler<unknown> | Array<Handler<unknown>> | StyleObject | UtilitiesConfig };

export type VariantsConfig = { [key: string]: string | string[] };

export interface Config<T extends object = Theme> {
  // presets?: Config[];
  // prefixer?: boolean;
  // attributify?: boolean | {
  //   prefix?: string
  //   separator?: string
  //   disable?: string[]
  // }
  separator?: string;
  important?: boolean | string;
  darkMode?: DarkModeConfig;
  theme?: T;
  // variantOrder?: string[];
  // plugins?: Plugin[];
  // handlers?: Handlers;
  // corePlugins?: (keyof BaseTheme)[] | string[] | { [ T in keyof BaseTheme ] : boolean } | { [ key:string ] : boolean };
  // prefix?: string;
  // exclude?: RegExp[];
  // alias?: { [key:string]: string };
  // shortcuts?: { [key:string]: Shortcut };
  // purge?: unknown;
  utilities?: UtilitiesConfig;
  variants?: VariantsConfig;
  styleLoader?: StyleLoader;
  [key:string]: any;
}

export function defineConfig (config: Config) {
  return config;
}
