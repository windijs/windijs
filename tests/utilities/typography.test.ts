import { createUtility } from "../../src";
import { fontFamilyConfig, fontSizeConfig } from "../../src/config/typography";
import { fontFamily, fontSize, fontStyle } from "../../src/utilities/typography";

test("Font Family", () => {
  const font = createUtility("font")
    .use(fontFamily(fontFamilyConfig))
    .init();
  expect(font.mono.css).toMatchSnapshot();
  expect(font.sans.css).toMatchSnapshot();
  expect(font.serif.css).toMatchSnapshot();
});

test("Font Size", () => {
  const text = createUtility("text")
    .use(fontSize(fontSizeConfig))
    .init();
  expect(text.xs.css).toMatchSnapshot();
  expect(text["2xl"].css).toMatchSnapshot();
  expect(text.base.css).toMatchSnapshot();
  expect(text.lg.css).toMatchSnapshot();
});

test("Font Size With Different Config", () => {
  const text = createUtility("text")
    .use(fontSize({
      xs: ["0.75rem", "1rem"],
      sm: ["0.875rem"],
      md: ["1rem", { lineHeight: "1.25rem", letterSpacing: "1px" }],
    }))
    .init();
  expect(text.xs.css).toMatchSnapshot();
  expect(text.sm.css).toMatchSnapshot();
  expect(text.md.css).toMatchSnapshot();
});

test("Font Style and Font Smoothing", () => {
  const { italic, antialiased } = createUtility("font").use(fontStyle()).init();
  // @ts-ignore safe entry ignored
  expect(italic.not.css).toMatchSnapshot();
  // @ts-ignore safe entry ignored
  expect(italic.css).toMatchSnapshot();
  // @ts-ignore safe entry ignored
  expect(antialiased.auto.css).toMatchSnapshot();
  // @ts-ignore safe entry ignored
  expect(antialiased.css).toMatchSnapshot();

  // @ts-ignore safe entry ignored
  expect(italic.meta).toMatchSnapshot();
  // @ts-ignore safe entry ignored
  expect(italic.not.meta).toMatchSnapshot();
  // @ts-ignore safe entry ignored
  expect(antialiased.meta).toMatchSnapshot();
  // @ts-ignore safe entry ignored
  expect(antialiased.auto.meta).toMatchSnapshot();
});
