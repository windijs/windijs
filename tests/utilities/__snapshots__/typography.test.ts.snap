// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Font Family 1`] = `
Object {
  "fontFamily": "ui-monospace,SFMono-Regular,Menlo,Monaco,Consolas,\\"Liberation Mono\\",\\"Courier New\\",monospace",
}
`;

exports[`Font Family 2`] = `
Object {
  "fontFamily": "ui-sans-serif,system-ui,-apple-system,BlinkMacSystemFont,\\"Segoe UI\\",Roboto,\\"Helvetica Neue\\",Arial,\\"Noto Sans\\",sans-serif,\\"Apple Color Emoji\\",\\"Segoe UI Emoji\\",\\"Segoe UI Symbol\\",\\"Noto Color Emoji\\"",
}
`;

exports[`Font Family 3`] = `
Object {
  "fontFamily": "ui-serif,Georgia,Cambria,\\"Times New Roman\\",Times,serif",
}
`;

exports[`Font Size 1`] = `
Object {
  "fontSize": "0.75rem",
  "lineHeight": "1rem",
}
`;

exports[`Font Size 2`] = `
Object {
  "fontSize": "1.5rem",
  "lineHeight": "2rem",
}
`;

exports[`Font Size 3`] = `
Object {
  "fontSize": "1rem",
  "lineHeight": "1.5rem",
}
`;

exports[`Font Size 4`] = `
Object {
  "fontSize": "1.125rem",
  "lineHeight": "1.75rem",
}
`;

exports[`Font Size With Different Config 1`] = `
Object {
  "fontSize": "0.75rem",
  "lineHeight": "1rem",
}
`;

exports[`Font Size With Different Config 2`] = `
Object {
  "fontSize": "0.875rem",
}
`;

exports[`Font Size With Different Config 3`] = `
Object {
  "fontSize": "1rem",
  "letterSpacing": "1px",
  "lineHeight": "1.25rem",
}
`;

exports[`Font Style and Font Smoothing 1`] = `
Object {
  "fontStyle": "italic",
}
`;

exports[`Font Style and Font Smoothing 2`] = `
Object {
  "fontStyle": "normal",
}
`;

exports[`Font Style and Font Smoothing 3`] = `
Object {
  "-moz-osx-font-smoothing": "grayscale",
  "-webkit-font-smoothing": "antialiased",
}
`;

exports[`Font Style and Font Smoothing 4`] = `
Object {
  "-moz-osx-font-smoothing": "auto",
  "-webkit-font-smoothing": "auto",
}
`;

exports[`Font Variant Numeric 1`] = `
Object {
  "fontVariantNumeric": "normal",
}
`;

exports[`Font Variant Numeric 2`] = `
Object {
  "fontVariantNumeric": "ordinal",
}
`;

exports[`Font Variant Numeric 3`] = `
Object {
  "fontVariantNumeric": "slashed-zero",
}
`;

exports[`Font Variant Numeric 4`] = `
Object {
  "fontVariantNumeric": "lining-nums",
}
`;

exports[`Font Variant Numeric 5`] = `
Object {
  "fontVariantNumeric": "oldstyle-nums",
}
`;

exports[`Font Variant Numeric 6`] = `
Object {
  "fontVariantNumeric": "proportional-nums",
}
`;

exports[`Font Variant Numeric 7`] = `
Object {
  "fontVariantNumeric": "tabular-nums",
}
`;

exports[`Font Variant Numeric 8`] = `
Object {
  "fontVariantNumeric": "diagonal-fractions",
}
`;

exports[`Font Variant Numeric 9`] = `
Object {
  "fontVariantNumeric": "stacked-fractions",
}
`;

exports[`Font Weight 1`] = `
Object {
  "fontWeight": "700",
}
`;

exports[`Font Weight 2`] = `
Object {
  "fontWeight": "500",
}
`;

exports[`Font Weight 3`] = `
Object {
  "fontStyle": "normal",
  "fontWeight": "400",
}
`;

exports[`Font Weight 4`] = `
Object {
  "fontStyle": "italic",
}
`;

exports[`Hyphens 1`] = `
Object {
  "-ms-hyphens": "manual",
  "-webkit-hyphens": "manual",
  "hyphens": "manual",
}
`;

exports[`Hyphens 2`] = `
Object {
  "-ms-hyphens": "auto",
  "-webkit-hyphens": "auto",
  "hyphens": "auto",
}
`;

exports[`Letter Spacing 1`] = `
Object {
  "letterSpacing": "-0.025em",
}
`;

exports[`Letter Spacing 2`] = `
Object {
  "letterSpacing": "0.05em",
}
`;

exports[`Line Height 1`] = `
Object {
  "lineHeight": "1rem",
}
`;

exports[`Line Height 2`] = `
Object {
  "lineHeight": "1.5",
}
`;

exports[`Line Height 3`] = `
Object {
  "lineHeight": "1.625",
}
`;

exports[`Tab Size 1`] = `
Object {
  "-moz-tab-size": "4",
  "-o-tab-size": "4",
  "tabSize": "4",
}
`;

exports[`Tab Size 2`] = `
Object {
  "-moz-tab-size": "2",
  "-o-tab-size": "2",
  "tabSize": "2",
}
`;

exports[`Text Alignment 1`] = `
Object {
  "textAlign": "left",
}
`;

exports[`Text Alignment 2`] = `
Object {
  "textAlign": "justify",
}
`;

exports[`Text Alignment 3`] = `
Object {
  "textAlign": "center",
}
`;

exports[`Text Alignment 4`] = `
Object {
  "textAlign": "right",
}
`;

exports[`Text Alignment 5`] = `
Object {
  "verticalAlign": "baseline",
}
`;

exports[`Text Alignment 6`] = `
Object {
  "verticalAlign": "top",
}
`;

exports[`Text Alignment 7`] = `
Object {
  "verticalAlign": "middle",
}
`;

exports[`Text Alignment 8`] = `
Object {
  "verticalAlign": "text-top",
}
`;

exports[`Text Alignment 9`] = `
Object {
  "verticalAlign": "text-bottom",
}
`;

exports[`Text Color 1`] = `
Object {
  "--w-text-opacity": "1",
  "color": "rgba(239, 68, 68, var(--w-text-opacity))",
}
`;

exports[`Text Color 2`] = `
Object {
  "--w-text-opacity": "0.1",
  "color": "rgba(59, 130, 246, var(--w-text-opacity))",
}
`;

exports[`Text Color 3`] = `
Object {
  "--w-text-opacity": "0.4",
}
`;

exports[`Text Decoration 1`] = `
Object {
  "-webkit-text-decoration-line": "underline",
  "textDecorationLine": "underline",
}
`;

exports[`Text Decoration 2`] = `
Object {
  "-webkit-text-decoration-line": "line-through",
  "textDecorationLine": "line-through",
}
`;

exports[`Text Decoration 3`] = `
Object {
  "-webkit-text-decoration-line": "overline",
  "textDecorationLine": "overline",
}
`;

exports[`Text Decoration 4`] = `
Object {
  "-webkit-text-decoration-line": "none",
  "textDecorationLine": "none",
}
`;

exports[`Text Decoration 5`] = `
Object {
  "-webkit-text-decoration-style": "solid",
  "textDecorationStyle": "solid",
}
`;

exports[`Text Decoration 6`] = `
Object {
  "-webkit-text-decoration-style": "wavy",
  "textDecorationStyle": "wavy",
}
`;

exports[`Text Decoration 7`] = `
Object {
  "textDecorationThickness": "2px",
}
`;

exports[`Text Decoration 8`] = `
Object {
  "textDecorationThickness": "auto",
}
`;

exports[`Text Decoration 9`] = `
Object {
  "--w-text-decoration-opacity": "1",
  "-webkit-text-decoration-color": "rgba(239, 68, 68, var(--w-text-decoration-opacity))",
  "textDecorationColor": "rgba(239, 68, 68, var(--w-text-decoration-opacity))",
}
`;

exports[`Text Decoration 10`] = `
Object {
  "--w-text-decoration-opacity": "0.2",
}
`;

exports[`Text Decoration 11`] = `
Object {
  "textUnderlineOffset": "3px",
}
`;

exports[`Text Indent 1`] = `
Object {
  "textIndent": "1.5rem",
}
`;

exports[`Text Indent 2`] = `
Object {
  "textIndent": "2rem",
}
`;

exports[`Text Indent 3`] = `
Object {
  "textIndent": "2.5rem",
}
`;

exports[`Text Shadow 1`] = `
Object {
  "textShadow": "0px 0px 1px rgb(0 0 0 / 20%), 0px 0px 1px rgb(1 0 5 / 10%)",
}
`;

exports[`Text Shadow 2`] = `
Object {
  "textShadow": "1px 1px 3px rgb(36 37 47 / 25%)",
}
`;

exports[`Text Shadow 3`] = `
Object {
  "textShadow": "none",
}
`;

exports[`Text Stroke 1`] = `
Object {
  "--w-text-stroke-opacity": "1",
  "-webkit-text-stroke-color": "rgba(252, 211, 77, var(--w-text-stroke-opacity))",
}
`;

exports[`Text Stroke 2`] = `
Object {
  "--w-text-stroke-opacity": "0.1",
  "-webkit-text-stroke-color": "rgba(147, 197, 253, var(--w-text-stroke-opacity))",
}
`;

exports[`Text Stroke 3`] = `
Object {
  "--w-text-stroke-opacity": "0.2",
}
`;

exports[`Text Stroke 4`] = `
Object {
  "-webkit-text-stroke-width": "thin",
}
`;

exports[`Text Stroke 5`] = `
Object {
  "-webkit-text-stroke-width": "0",
}
`;

exports[`Text Stroke 6`] = `
Object {
  "-webkit-text-stroke-width": "medium",
}
`;

exports[`Text Stroke 7`] = `
Object {
  "-webkit-text-stroke-width": "thick",
}
`;

exports[`Text Stroke 8`] = `
Object {
  "-webkit-text-stroke-width": "5px",
}
`;

exports[`Text transform 1`] = `
Object {
  "textTransform": "none",
}
`;

exports[`Text transform 2`] = `
Object {
  "textTransform": "uppercase",
}
`;

exports[`Text transform 3`] = `
Object {
  "textTransform": "capitalize",
}
`;

exports[`Whitespace 1`] = `
Object {
  "whiteSpace": "normal",
}
`;

exports[`Whitespace 2`] = `
Object {
  "whiteSpace": "nowrap",
}
`;

exports[`Whitespace 3`] = `
Object {
  "whiteSpace": "pre",
}
`;

exports[`Whitespace 4`] = `
Object {
  "whiteSpace": "pre-line",
}
`;

exports[`Whitespace 5`] = `
Object {
  "whiteSpace": "pre-wrap",
}
`;

exports[`WordBreak 1`] = `
Object {
  "wordBreak": "break-all",
}
`;

exports[`WordBreak 2`] = `
Object {
  "overflowWrap": "break-word",
}
`;

exports[`WordBreak 3`] = `
Object {
  "overflowWrap": "normal",
  "wordBreak": "normal",
}
`;

exports[`Writing Mode & Text Orientation 1`] = `
Object {
  "-ms-writing-mode": "lr-tb",
  "-webkit-writing-mode": "horizontal-tb",
  "writingMode": "horizontal-tb",
}
`;

exports[`Writing Mode & Text Orientation 2`] = `
Object {
  "-ms-writing-mode": "tb-rl",
  "-webkit-writing-mode": "vertical-rl",
  "writingMode": "vertical-rl",
}
`;

exports[`Writing Mode & Text Orientation 3`] = `
Object {
  "-ms-writing-mode": "tb-lr",
  "-webkit-writing-mode": "vertical-lr",
  "writingMode": "vertical-lr",
}
`;

exports[`Writing Mode & Text Orientation 4`] = `
Object {
  "-webkit-text-orientation": "mixed",
  "textOrientation": "mixed",
}
`;

exports[`Writing Mode & Text Orientation 5`] = `
Object {
  "-webkit-text-orientation": "upright",
  "textOrientation": "upright",
}
`;

exports[`Writing Mode & Text Orientation 6`] = `
Object {
  "-webkit-text-orientation": "sideways",
  "textOrientation": "sideways",
}
`;
